### DISCUSSION TITLE
Opencaesar connected to a fusaki server on AWS?

### DISCUSSION THREAD

**Discussion by PatrickF-mbse** [opencaesar/platform#4] [General]
Hi

I was wondering if anyone is using Opencaesar connected to a server located on Amazon cloud or Google cloud? I am process of learning the installation process with eclipse and I think if I understood well the database is on the same computer than the installation. 

I would like to have it on the cloud to enable multi user queries and run some transformation services automatically. The eclipse clients could connect to it to edit the views/diagrams/tables.

 Do you think this is quite easy to do? Interested to know if anybody tried a similar setup.

Thank you

**Comment by melaasar** [comment]
Hi Patric. See blow.

> I was wondering if anyone is using Opencaesar connected to a server located on Amazon cloud or Google cloud?

Are you talking about a triple store (database) for RDF/OWL data hosted on these platforms, like the Neptune service on AWS?

> I am process of learning the installation process with eclipse and I think if I understood well the database is on the same computer than the installation.

The ability to start/stop a local Fuseki database is not provided by the Eclipse-based Rosetta WB per se. It is rather a capability of the Gradle script you get when you create an OML project. You can choose to invoke such capability form the Eclipse UI, or you can invoke it from a terminal using the gradlew CLI. This is provided for convenience only. You can configure it to have persistent TDB database (instead of in-memory database). You can also configure it to have web UI. You may choose instead to work with a Fuseki instance already deployed on another server, or work a different RDF triple store altogether, like Neptune, GraphDB, etc. This part can be very configurable to suit your operational needs.

>I would like to have it on the cloud to enable multi user queries and run some transformation services automatically. 

Differentiate between the authoring use cases enabled by the OML files, and the analysis use cases enabled by the RDF/OWL triple store. The authoring can be done in Eclipse (Rosetta), VS Code (Luxor) or other IDEs. It is done by modifying the OML files, either with a text editor, with UI viewpoints, or with APIs. When you want to do analysis, you build the OML project and publish the models to a RDF Triple Store. This can then enable the "analysis" use cases like making queries, creating visualizations, running transformations, etc. The Gradle script provided in an OML project performs such analysis in a non-interactive way meant to be orchestrated by a CI process. This is why it launches a transient Fuseki instance However, if you published the models to a persistent Fuseki instance (or another triple store), you can send your queries interactively and by multiple users. Fuseki can work in this mode (just needs to be configured this way), or you can use other triple store as stated above.

>The eclipse clients could connect to it to edit the views/diagrams/tables.

Do not confuse the Eclipse views that are used for authoring OML files, and similar analysis views you can generate from the RFD database, which tend to be web-based views that can run within or outside Eclipse (like on a web server).
